src/components/Widgets/DateTimeControl.js

```diff
@@ -1,20 +1,50 @@
 import PropTypes from 'prop-types';
 import React from 'react';
 import DateTime from 'react-datetime';
+import ImmutablePropTypes from 'react-immutable-proptypes';
 
 export default class DateTimeControl extends React.Component {
-  componentDidMount() {
-    if (!this.props.value) {
-      this.props.onChange(new Date());
-    }
+  constructor(props) {
+    super(props);
+    this.state = { value: this.getStateValue(this.props.value) };
   }
 
   handleChange = (datetime) => {
     this.props.onChange(datetime);
   };
 
+  getStateValue(value) {
+    if (value === false || value === '') {
+      return '';
+    } else if(!value) {
+      return new Date();
+    } else {
+      return value;
+    }
+  }
+
+  componentWillReceiveProps(props) {
```

Comment:
We shouldn't need this or internal state - just run the `getStateValue` transformer on the value received through props (probably rename the function, though). Any reason that wouldn't work?

Reply:
I think I was just overengineering :-p

It works well :-)